src/main/java/org/bankofspring/model/Account.java

```diff
@@ -0,0 +1,136 @@
+package org.bankofspring.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class to represent a bank account
+ * we have added a maxBalanceAmount to maintain the maximum allowed for an account
+ * note that an Account has a bidirectional relationship with customer and that
+ * more than one customer can be linked to an account (joined account)
+ * we have added facilities to add customer against the account but no facilities as yet to 
+ * remove an customer from account
+ */
+public class Account {
+	
+	private String accountNumber;
+	private String accountDescription;
+	private long accountBalance; 
+	private long maxBalanceAmount;
+	
+	//List of owning customers, for e.g., joint accounts
+	private List<Customer> owningCustomers = new ArrayList<Customer>();
+	private List<AccountTransaction> transactions = new ArrayList<AccountTransaction>();;
+	
+	public Account(String accountNumber, String accountDescription, List<Customer> owningCustomers){
+		setAccountNumber(accountNumber);
+		setAccountDescription(accountDescription);
+		setOwningCustomers(owningCustomers);
+		setAccountBalance(0L);
+		setMaxBalanceAmount(Long.MAX_VALUE); //Make very high unless specified 
+	}
+
+	public String getAccountNumber() {
+		return accountNumber;
+	}
+
+	public void setAccountNumber(String accountNumber) {
+		this.accountNumber = accountNumber;
+	}
+
+	public String getAccountDescription() {
+		return accountDescription;
+	}
+
+	public void setAccountDescription(String accountDescription) {
+		this.accountDescription = accountDescription;
+	}
+
+	public long getAccountBalance() {
+		return accountBalance;
+	}
+
+	public void setAccountBalance(long accountBalance) {
+		this.accountBalance = accountBalance;
+	}
```

Comment:
This seems controversial given that a lot of people didn't like the idea of having a publically-accessible method to set account balance. Opinions, people?